### PR TITLE
Remove NonStop and Tandem

### PR DESCRIPTION
**Pull request by richsalz**
## Checklist
Thank you for your contribution. Please answer the following:

- [x] I acknowledge that I am authorized to submit this code under
the terms of the [Apache License](https://www.apache.org/licenses/LICENSE-2.0)
- [ ] All new public APIs are documented
- [ ] All new public functions have tests
- [ ] All new public functions have fuzzing tests

### CODE REVIEW

File: crypto/ui/ui_openssl.c [hunk 1]
```diff
@@ -127,10 +127,6 @@
 #  define TTY_set(tty,data)      ioctl(tty,TIOCSETP,data)
 # endif
 
-# if !defined(_LIBC) && !defined(OPENSSL_SYS_MSDOS) && !defined(OPENSSL_SYS_VMS) && ! (defined(OPENSSL_SYS_TANDEM) && defined(_SPT_MODEL_))
```
Comment (botovq) [inline crypto/ui/ui_openssl.c:130]: I think this should stay, although the VMS bit could probably also go.

```
# if !defined(_LIBC) && !defined(OPENSSL_SYS_MSDOS) && !defined(OPENSSL_SYS_VMS)
```

**Review by botovq — DISMISSED** [review]
Only one thing here, I am not sure that we can rely on `<termio.h>` and `<sgtty.h>` pulling in `<sys/ioctl.h>` for the `ioctl()` calls for `TTY_get()` and `TTY_set()`.

**Comment by botovq** [comment]
I missed a couple of things:

https://github.com/quictls/quictls/blob/52c2d76e8b3561701ebec918ade682648939ab4f/NOTES-PERL.md?plain=1#L7
https://github.com/quictls/quictls/blob/52c2d76e8b3561701ebec918ade682648939ab4f/NOTES-PERL.md?plain=1#L55-L65
https://github.com/quictls/quictls/blob/52c2d76e8b3561701ebec918ade682648939ab4f/util/shlib_wrap.sh.in#L87-L95
https://github.com/quictls/quictls/blob/52c2d76e8b3561701ebec918ade682648939ab4f/test/recipes/01-test_symbol_presence.t#L24
https://github.com/quictls/quictls/blob/52c2d76e8b3561701ebec918ade682648939ab4f/test/recipes/90-test_fipsload.t#L22
https://github.com/quictls/quictls/blob/rm-old-hp-os/test/recipes/90-test_shlibload.t#L23

File: crypto/ui/ui_openssl.c (hunk 1 above)
Comment (richsalz) [reply crypto/ui/ui_openssl.c:130]: I have a fixup commit that puts the ioctl.h include inside the termio and sgtty cases, and restores the conio include.  Please take a look.

**Comment by richsalz** [comment]
Added a commit to catch the things you found that I missed.

File: crypto/ui/ui_openssl.c [hunk 2]
```diff
@@ -112,6 +112,7 @@
 # endif
 
 # ifdef TERMIO
+#  include <ioctl.h>
```
Comment (botovq) [inline crypto/ui/ui_openssl.c:115]:
```suggestion
#  include <sys/ioctl.h>
```

File: crypto/ui/ui_openssl.c [hunk 3]
```diff
@@ -120,14 +121,15 @@
 # endif
 
 # ifdef SGTTY
+#  include <ioctl.h>
```
Comment (botovq) [inline crypto/ui/ui_openssl.c:124]:
```suggestion
#  include <sys/ioctl.h>
```

**Review by botovq — COMMENTED** [review]
Looks good. There's still
https://github.com/quictls/quictls/blob/52c2d76e8b3561701ebec918ade682648939ab4f/util/shlib_wrap.sh.in#L87-L95

**Comment by richsalz** [comment]
Fixup that fixes the include for `ioctl.h` and removes the lines from `shlib_wrap.sh.in`

**Review by botovq — DISMISSED** [review]

**Comment by richsalz** [comment]
Had a merge conflict (CHANGES.md) need you to re-approve, @botovq

**Review by botovq — APPROVED** [review]

**Pull request merged**